packages/@aws-cdk/aws-ecs/lib/base/task-definition.ts

```diff
@@ -175,10 +185,55 @@ export interface TaskDefinitionProps extends CommonTaskDefinitionProps {
   readonly pidMode?: PidMode;
 }
 
+/**
+ * The common task definition attributes used across all types of task definitions.
+ */
+export interface CommonTaskDefinitionAttributes {
+  /**
+   * The arn of the task definition
+   */
+  readonly taskDefinitionArn: string;
+
+  /**
+   * The networking mode to use for the containers in the task.
+   *
+   * @default NetworkMode.BRIDGE
+   */
+  readonly networkMode?: NetworkMode;
+
+  /**
+   * The name of the IAM role that grants containers in the task permission to call AWS APIs on your behalf.
+   *
+   * @default undefined.
+   */
+  readonly taskRole?: iam.IRole;
+}
+
+/**
+ *  A reference to an existing task definition
+ */
+export interface TaskDefinitionAttributes extends CommonTaskDefinitionAttributes {
+  /**
+   * Execution role for this task definition
+   *
+   * @default: undefined
+   */
+  readonly executionRole?: iam.IRole;
```

Comment:
I don't understand why `executionRole` is here, and not in the EC2 TaskDefinitionAttributes 🤷‍♂️. An oversight on our part?

Reply:
Good catch. We might be able to get rid of `executionRole` here. 

`fromTaskDefinitionArn` returns an object containing `executionRole` to satisfy its parent class, TaskDefinitionBase, which [containing executionRole?](https://github.com/aws/aws-cdk/blob/98ab29206903a235331a9a0b4b38a5e6ad6d89dc/packages/%40aws-cdk/aws-ecs/lib/base/task-definition.ts#L182).

I was originally hoping to give users flexibility to define `executionRole` by themselves, but considering these methods are to import from existing resources, users might actually hope to give as minimal attributes as possible. I probably should keep `executionRole` to be `undefined` and drop it from `TaskDefinitionAttributes`.

Thoughts?

Reply:
> I probably should keep `executionRole` to be `undefined` and drop it from `TaskDefinitionAttributes`.

Agreed. iirc `executionRole` only makes sense for EC2 taskdefinitions (and is only on `Base` to satisfy some interface, but is never set there). For now, I think it's fine to take it out altogether.